source/CMakeLists.txt

```diff
@@ -78,8 +78,12 @@ file(GLOB sources ${PROJECT_SOURCE_DIR}/source/src/*.cxx)
 file(GLOB headers ${PROJECT_SOURCE_DIR}/source/include/*.h)
 
 #---Add library-----------------------------------------------------------------
+set(ROOT_DEPS ROOT::Core ROOT::RIO  ROOT::Rint ROOT::Matrix ROOT::Physics ROOT::MathCore
```

Comment:
There is no need to define ROOT_DEPS. Once we added  Geom, EG into the required components, ${ROOT_LIBRARIES} include all needed ROOT libraries, so we can remove this additional variable and then change the  line 
`target_link_libraries(${library_name} ${ROOT_LIBRARIES} EG Geom)`
to
`target_link_libraries(${library_name} ${ROOT_LIBRARIES})`

Reply:
In my opinion we should keep `${ROOT::DEPS}` as it defines the ROOT libraries via their name. If you specify e.g. `${ROOT_LIBRARIES}` in
`set_target_properties(${library_name} PROPERTIES INTERFACE_LINK_LIBRARIES "${ROOT_LIBRARIES}")`
you would end up with hard-coded library paths in the CMake target file which is exactly what we don't want as we cannot really use pre-built binaries. As an example the VMCTarget.cmake contains:

using `${ROOT_LIBRARIES}`: `INTERFACE_LINK_LIBRARIES "/home/bvolkel/work/software/aliceDev/VMCReplay/test/install/root/lib/libCore.so`

vs.

using `${ROOT_DEPS}`: INTERFACE_LINK_LIBRARIES "ROOT::Core"

In the latter case, if we have found ROOT e.g. in O2, we can access the libraries via their "aliases" (also if VMC was pre-built as is is done for some packages in aliBuild). In general I think it's a good feature to be able to be used from pre-built?!

But let me think about it, maybe there is a way in between or I find another CMake feature to make it work consistently and I will come back to you tonight or tomorrow.

My other question would be: Why do we link against all default ROOT libraries if we know what we need? I think it would not hurt in the end but in the target file it looks like we need a lot of libraries but we don't need many of them (ROOTDataFrame, Tree, Graf, Net, Postscript, TreePlayer and some others)